datacube/drivers/postgis/_api.py

```diff
@@ -663,12 +666,21 @@ def search_datasets(self, expressions,
         :type with_source_ids: bool
         :type select_fields: tuple[datacube.drivers.postgis._fields.PgField]
         :type expressions: tuple[datacube.drivers.postgis._fields.PgExpression]
+
+        :return: An iterable of tuples of decoded values
         """
+        if select_fields is None:
+            select_fields = _dataset_fields()
         select_query = self.search_datasets_query(expressions, source_exprs,
                                                   select_fields, with_source_ids,
                                                   limit, geom=geom, archived=archived)
         _LOG.debug("search_datasets SQL: %s", str(select_query))
-        return self._connection.execute(select_query)
+
+        def decode_row(raw):
```

Comment:
Can this (new) function get a  type signature?

Reply:
Yep.